### ISSUE TITLE
VK_KHR_maintenance1 removes VK_KHR_push_descriptor VU

### ISSUE THREAD

**Issue by krOoze** [KhronosGroup/Vulkan-Docs#573]
`VK_KHR_maintenance1` removes `VkDescriptorSetAllocateInfo` VU of `VK_KHR_push_descriptor` (in addition to Issue #533):

```
ifndef::VK_KHR_maintenance1[]

.Valid Usage
****
  * [[VUID-VkDescriptorSetAllocateInfo-descriptorSetCount-00306]]
    pname:descriptorSetCount must: not be greater than the number of sets
    that are currently available for allocation in pname:descriptorPool
  * [[VUID-VkDescriptorSetAllocateInfo-descriptorPool-00307]]
    pname:descriptorPool must: have enough free descriptor capacity
    remaining to allocate the descriptor sets of the specified layouts
ifdef::VK_KHR_push_descriptor[]
  * [[VUID-VkDescriptorSetAllocateInfo-pSetLayouts-00308]]
    Any given element of pname:pSetLayouts must: not have been created with
    ename:VK_DESCRIPTOR_SET_LAYOUT_CREATE_PUSH_DESCRIPTOR_BIT_KHR set
endif::VK_KHR_push_descriptor[]
****
endif::VK_KHR_maintenance1[]
```

Is that intended or bug?
If intended it should probably fail anyway; which `VK_ERROR_*` would cover that case?

**Comment by pdaniell-nv** [comment]
Yes, this was intentional because when VK_KHR_maintenance1 was added along with it came VK_ERROR_OUT_OF_POOL_MEMORY_KHR which requires the implementation to verify the descriptor size can be satisfied rather than the validation layers.

**Comment by krOoze** [comment]
@pdaniell-nv Soo, `pSetLayouts` created with `VK_DESCRIPTOR_SET_LAYOUT_CREATE_PUSH_DESCRIPTOR_BIT_KHR` results in what?
Sounds like a logical error to me and so more suited to have a VU.

**Comment by pdaniell-nv** [comment]
You're right, that part should remain. I'll fix this internally.

**Comment by oddhack** [comment]
This should be fixed in the 1.0.62 spec update.
